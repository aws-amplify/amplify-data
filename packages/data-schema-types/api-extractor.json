{
  "extends": "../../api-extractor.base.json",
  "mainEntryPointFilePath": "<projectFolder>/dist/esm/index.d.ts"
}
