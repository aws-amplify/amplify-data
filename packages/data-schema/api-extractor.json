{
  "extends": "../../api-extractor.base.json",
  "mainEntryPointFilePath": "<projectFolder>/dist/esm/src/index.d.ts"
}
