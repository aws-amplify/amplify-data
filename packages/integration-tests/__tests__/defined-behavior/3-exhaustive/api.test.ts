import * as path from 'path';
import * as fs from 'fs';
import * as crypto from 'crypto';
import { rimraf } from 'rimraf';
import { Extractor, ExtractorConfig } from '@microsoft/api-extractor';
import { execSync } from 'child_process';

/**
 * Where api-extractor drops its summary of the API.
 *
 * This is just the default path. Changing this variable alone will not current cause
 * api-extractor to output to a different folder. If we need to change this, refer to
 * `integration-tests/api-extractor.json` to find the config keys that will need to
 * be changed and/or overridden in the config.
 */
const TEMP_EXTRACTION_PATH = path.join(__dirname, '../../../temp');

/**
 * Where the latest/current docs are stored.
 */
const API_DOCS_PATH = path.join(__dirname, '../../../docs');

describe('Publicly exposed API and docstrings', () => {
  test(
    'has not changed',
    async () => {
      /**
       * This builds the API extraction JSON, which is what the documenter will
       * need to consume.
       */
      const apiExtractorJsonPath: string = path.join(
        __dirname,
        '../../../api-extractor.json',
      );
      Extractor.invoke(
        ExtractorConfig.loadFileAndPrepare(apiExtractorJsonPath),
        {
          // Equivalent to the "--local" command-line parameter
          localBuild: true,
        },
      );

      /**
       * Clobber and re-generate user-facing API docs.
       *
       * Programmatic access to api-documentor is non-obvious. So, path of least
       * resistence there is to just call via the CLI.
       */
      await rimraf(API_DOCS_PATH);
      execSync(
        `npx api-documenter markdown --input-folder ${TEMP_EXTRACTION_PATH} --output-folder ${API_DOCS_PATH}`,
      );

      /**
       * Confirm there hasn't been a change by comparing a hash of this latest
       * docs generation to the previous one.
       */
      const docFiles = fs.readdirSync(API_DOCS_PATH);
      const computedHash = crypto.createHash('sha256');
      for (const fileName of docFiles) {
        const fileData = fs
          .readFileSync(`${API_DOCS_PATH}/${fileName}`)
          .toString();
        computedHash.update([fileName, fileData].join('\n'));
      }
      expect(computedHash.digest('hex')).toMatchSnapshot();
    },
    30 * 1000,
  );
});
