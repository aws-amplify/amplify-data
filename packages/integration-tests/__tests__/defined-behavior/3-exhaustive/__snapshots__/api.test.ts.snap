// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publicly exposed API and docstrings have not changed 1`] = `"a0b41e261f533a7d243d31c3fad8429717024a767d17daa118f8157d2afcc967"`;
